src/vs/platform/theme/common/colorRegistry.ts

```diff
@@ -215,7 +215,9 @@ export const foreground = registerColor('foreground', { dark: '#CCCCCC', light:
 export const disabledForeground = registerColor('disabledForeground', { dark: '#CCCCCC80', light: '#61616180', hcDark: '#A5A5A5', hcLight: '#7F7F7F' }, nls.localize('disabledForeground', "Overall foreground for disabled elements. This color is only used if not overridden by a component."));
 export const errorForeground = registerColor('errorForeground', { dark: '#F48771', light: '#A1260D', hcDark: '#F48771', hcLight: '#B5200D' }, nls.localize('errorForeground', "Overall foreground color for error messages. This color is only used if not overridden by a component."));
 export const descriptionForeground = registerColor('descriptionForeground', { light: '#717171', dark: transparent(foreground, 0.7), hcDark: transparent(foreground, 0.7), hcLight: transparent(foreground, 0.7) }, nls.localize('descriptionForeground', "Foreground color for description text providing additional information, for example for a label."));
-export const iconForeground = registerColor('icon.foreground', { dark: '#C5C5C5', light: '#424242', hcDark: '#FFFFFF', hcLight: '#292929' }, nls.localize('iconForeground', "The default color for icons in the workbench."));
+
+// {{SQL CARBON EDIT}} -- Update icon foreground color and let it be set by theme
+export const iconForeground = registerColor('icon.foreground', { dark: null, light: null, hcDark: null, hcLight: null }, nls.localize('iconForeground', "The default color for icons in the workbench."));
```

Comment:
Why are we using the icon foreground color for text? Shouldn't we be using the same color text as the rest of the dialog regardless of what the icon colors are like?

Reply:
This modifies the style 'color' on label as 'codicon' class is applied on the label. I checked almost all themes, and these hard-coded colors don't look too different than the theme defaults, so we shouldn't need to set them here.

Reply:
We probably shouldn't be using codicon for the entire row (label + icon). The label isn't an icon after all. I would suggest looking into fixing that first. Generally, the label and icon would be separate elements in the DOM (usually as siblings in a div together).

If we're using the icon color for these labels then if someone goes and changes the color of icons to something completely different the labels will also show up with that different color (but not the other labels on the dialog)

Reply:
I couldn't make it work to get desired alignment without using background image, and couldn't find any other example for reference. If you have a solution for reference, please let me know.

Reply:
The modelview dialogs have a text tooltip that can be set - e.g. from the deploy BDC wizard : 

![image](https://user-images.githubusercontent.com/28519865/203182209-ede0c87b-a7f0-415c-b471-812839c9711d.png)

This is done in code in src\sql\workbench\browser\modelComponents\text.component.ts - look at the modelview-text-tooltip.

Although you'll still need to be using background-image regardless, it's just that it'll be on its own element (that's a sibling of the text label element) instead of being a class directly on the label element itself.

![image](https://user-images.githubusercontent.com/28519865/203182526-6e292228-4f47-4ec6-9acf-9b5b8bf343ff.png)

Reply:
It seems like the 'codicon' class is not applied here, if that's what we are okay with, I can remove this class itself from the component. We don't need to change the whole design.

Reply:
Also to mention, icon colors don't change with CSS, its defined in svg files, so this infocon color is a myth.

Reply:
Apparently, I cannot remove codicon css, as it's needed for this design, I this the change is fine as-is as the icon-foreground color applies only to text on foreground and not the icon itself. And because all our text should be of same color, this change is probably necessary for proper theming.

Reply:
Apparently, removing 'codicon' breaks css consistency in common-icons.css, so I wouldn't recommend we go for it.

> If we're using the icon color for these labels then if someone goes and changes the color of icons to something completely different the labels will also show up with that different color (but not the other labels on the dialog)

And as mentioned above, this is not the case with codicons, I don't think we should have any concern.

Reply:
That's for our icons - codicons are a VS Code core concept and are done with fonts so they inherit the colors. (we've just hacked in our mostly-SVG based icons to use the same class for consistency and ease of use but with the downside that they don't get affected by settings such as the icon color) 

See https://code.visualstudio.com/api/references/theme-color#base-colors

For example, if you put this in your settings.json then it will color the icons red : 

```json
	"workbench.colorCustomizations": {
		"icon.foreground": "#ff0000"
	}
```

![image](https://user-images.githubusercontent.com/28519865/203191379-de1cdc8c-9594-403b-b11a-1f0bf6fc9be4.png)

which then results in this for the labels

![image](https://user-images.githubusercontent.com/28519865/203191576-db96b1d1-a48e-43fe-96ad-2a96f33c2cca.png)


(doesn't really matter here, but FYI you can also change the colors of SVGs with CSS - see https://qawithexperts.com/article/html/how-to-change-color-of-svg-various-ways-using-css/414 for some examples)